flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionCapacitySchedulerITCase.java

```diff
@@ -445,39 +444,51 @@ private static Map<String, String> getFlinkConfig(final String host, final int p
                     LOG.info("Starting testNonexistingQueueWARNmessage()");
-                    try {
-                        runWithArgs(
-                                new String[] {
-                                    "-j",
-                                    flinkUberjar.getAbsolutePath(),
-                                    "-t",
-                                    flinkLibFolder.getAbsolutePath(),
-                                    "-jm",
-                                    "768m",
-                                    "-tm",
-                                    "1024m",
-                                    "-qu",
-                                    "doesntExist"
-                                },
-                                "to unknown queue: doesntExist",
-                                null,
-                                RunTypes.YARN_SESSION,
-                                1);
-                    } catch (Exception e) {
-                        assertTrue(
-                                ExceptionUtils.findThrowableWithMessage(
-                                                e, "to unknown queue: doesntExist")
-                                        .isPresent());
-                    }
-                    assertThat(
-                            yarTestLoggerResource.getMessages(),
-                            hasItem(
-                                    containsString(
-                                            "The specified queue 'doesntExist' does not exist. Available queues")));
+                    assertThatThrownBy(
+                                    () ->
+                                            runWithArgs(
+                                                    new String[] {
+                                                        "-j",
+                                                        flinkUberjar.getAbsolutePath(),
+                                                        "-t",
+                                                        flinkLibFolder.getAbsolutePath(),
+                                                        "-jm",
+                                                        "768m",
+                                                        "-tm",
+                                                        "1024m",
+                                                        "-qu",
+                                                        "doesntExist"
+                                                    },
+                                                    "to unknown queue: doesntExist",
+                                                    null,
+                                                    RunTypes.YARN_SESSION,
+                                                    1))
+                            .isInstanceOf(Exception.class)
+                            .getRootCause()
+                            .hasMessageContaining("to unknown queue: doesntExist");
```

Comment:
this is a bit different. Previously we accepted the exception at any place in the cause chain. Could use satifies(FlinkAssertions...) instead

Reply:
@zentol  Do you think I could use original `ExceptionUtils.findThrowableWithMessage` to instead `FlinkAssertions` ?

Reply:
Why don't you want to use `FlinkAssertions#anyCauseMatches`?

Reply:
Done~.